packages/product/integration-tests/__tests__/services/product-category/index.ts

```diff
@@ -173,6 +170,365 @@ moduleIntegrationTestRunner({
+              mpath: "category-0.category-1.category-1-b.",
+              parent_category_id: "category-1",
+              parent_category: {
+                id: "category-1",
+                handle: "category-1",
+                mpath: "category-0.category-1.",
+                parent_category_id: "category-0",
+                parent_category: {
+                  id: "category-0",
+                  handle: "category-0",
+                  mpath: "category-0.",
+                  parent_category_id: null,
+                  parent_category: null,
+                },
+              },
+            },
+          ])
+        })
+
+        it("includes the entire list of descendants an parents when include_descendants_tree and include_parents_tree are true for multiple results", async () => {
+          const productCategoryResults = await service.list(
+            {
+              parent_category_id: "category-1",
+              include_descendants_tree: true,
+              include_parents_tree: true,
+            },
+            {
+              select: ["id", "handle"],
+            }
+          )
+
+          const serializedObject = JSON.parse(
+            JSON.stringify(productCategoryResults)
+          )
+
+          expect(serializedObject).toEqual([
+            {
+              id: "category-1-a",
+              handle: "category-1-a",
+              mpath: "category-0.category-1.category-1-a.",
+              parent_category_id: "category-1",
+              parent_category: {
+                id: "category-1",
+                handle: "category-1",
+                mpath: "category-0.category-1.",
+                parent_category_id: "category-0",
+                parent_category: {
+                  id: "category-0",
+                  handle: "category-0",
+                  mpath: "category-0.",
+                  parent_category_id: null,
+                  parent_category: null,
+                },
+              },
+              category_children: [],
+            },
+            {
+              id: "category-1-b",
+              handle: "category-1-b",
+              mpath: "category-0.category-1.category-1-b.",
+              parent_category_id: "category-1",
+              parent_category: {
+                id: "category-1",
+                handle: "category-1",
+                mpath: "category-0.category-1.",
+                parent_category_id: "category-0",
+                parent_category: {
+                  id: "category-0",
+                  handle: "category-0",
+                  mpath: "category-0.",
+                  parent_category_id: null,
+                  parent_category: null,
+                },
+              },
+              category_children: [
+                {
+                  id: "category-1-b-1",
+                  handle: "category-1-b-1",
+                  mpath: "category-0.category-1.category-1-b.category-1-b-1.",
+                  parent_category_id: "category-1-b",
```

Comment:
shouldn't there be an empty category_children here?
```suggestion
                  parent_category_id: "category-1-b",
                  category_children: [],
```

Reply:
When both flags are `true` we decided to remove the parent and children properties from categories that are not in the root level. This is tricky to support without dealing with circular references.